ci/main.py

```
@@ -3,61 +3,130 @@
+                    build_number = 0
+                if is_already_uploaded(name, version, build_number, channel):
+                    # Only new packages (either version or build_number)
+                    log.info("Skipping package: {0}-{1}".format(name, version))
+                    continue
                 build(name, version, root)
                 if os.environ['TRAVIS_SECURE_ENV_VARS'] == 'true':
-                    upload(name, version)
+                    upload(name, version, channel)
                 else:
-                    print("Uploading not available in Pull Requests")
+                    log.info("Uploading not available in Pull Requests")
 
 
 def build(name, version, root):
-    print("Building package: {0}-{1}".format(name, version))
+    '''Build a recipe.
+
+    Parameters
+    ----------
+    root : str
+        the directory path for the recipe.
+    '''
     # Quote is need in case the root path has spaces in it.
     build_cmd = 'conda build "%s"' % root
-    print('BUILDING COMMAND: {0}'.format(build_cmd))
+    log.info('Build: {0}'.format(build_cmd))
     check_call(build_cmd, shell=True)
 
 
-def upload(name, version):
+def is_already_uploaded(name, version, build_number, channel):
+    '''Check if we want to build & upload a package.
+
+    It checks package name, version and build number
+    sequentially to decide whether to build and upload it or not.
+
+    Parameters
+    ----------
+    name : str
+        Package name.
+    version : str
+        Package version.
+    build_number : int
+        Build number
+    channel : str
+        Anaconda channel to check the previous build.
+
+    Returns
+    -------
+    Bool
+        If a package in the channel has the same name, the same
+        version, and an equal or higher build number, then return
+        True; otherwise, return False.
+
+    '''
     check_cmd = ('conda search --json --override-channels '
                  '-c {0} --spec {1}={2}').format(
```

this is wrong if I remember correctly. You can't specify version here. If it is not found, it will raise error instead of return `{}`